server/src/main/java/org/elasticsearch/action/support/IndicesOptions.java

```diff
@@ -227,8 +237,21 @@ public boolean ignoreThrottled() {
         return EnumSet.copyOf(expandWildcards);
     }
 
+    /**
+     * @return Whether to include data streams when resolving index expressions to concrete indices.
+     */
+    public boolean includeDataStreams() {
+        return options.contains(Option.INCLUDE_DATA_STREAMS);
+    }
+
     public void writeIndicesOptions(StreamOutput out) throws IOException {
-        out.writeEnumSet(options);
+        if (out.getVersion().before(Version.V_8_0_0) && options.contains(Option.INCLUDE_DATA_STREAMS)) {
+            EnumSet<Option> copy = EnumSet.copyOf(options);
+            copy.remove(Option.INCLUDE_DATA_STREAMS);
```

Comment:
I think we should put in a check into CreateDataStreamAction to not allow creating data streams before all nodes are on the target release (7.8)? To ensure that there are no surprises in a mixed cluster.

Reply:
👍 I will make this change in a followup change.